pkg/controlplane/controller/leaderelection/run_with_leaderelection.go

```diff
@@ -36,11 +36,10 @@ type NewRunner func() (func(ctx context.Context, workers int), error)
 // RunWithLeaderElection only returns when the context is done, or initial
 // leader election fails.
 func RunWithLeaderElection(ctx context.Context, config *rest.Config, newRunnerFn NewRunner) {
-	var cancel context.CancelFunc
+	ctx, cancel := context.WithCancel(ctx)
 
 	callbacks := leaderelection.LeaderCallbacks{
 		OnStartedLeading: func(ctx context.Context) {
-			ctx, cancel = context.WithCancel(ctx)
```

Comment:
We need to use the context created on L42 instead of the context on L38 so we can't change that.

Reply:
Got it. So we just need to check if cancel is not nil, correct?

Reply:
Yeah this is the fix I imagined, though I'm wondering if we have an easy way of testing whether the fix works. I think the integration test that fails is flaky so is not reproducible that frequently.